tests/e2e/matmul/generate_e2e_matmul_tests.py

```diff
@@ -155,7 +157,7 @@ def get_dynamicities(shapes_id: ShapesId):
 
 @dataclasses.dataclass
 class TileWorkgroupSizePair:
-  tile_size: typing.List[int]
+  tile_size: typing.Union[typing.List[int], typing.List[typing.List[int]]]
```

Comment:
Instead of doing this union and the if-else cases in the following, let's change the definition of `TileWorkgroupSizePair` to use `typing.List[typing.List[int]]` for `tile_size` and update other cases. This would avoid quite some complexity.

Reply:
Sure sounds good. I have updated the python file.